src/modules/LanItems.rb

```diff
@@ -1817,6 +1803,31 @@ def setup_basic_device_options(devmap)
       devmap
     end
 
+    # Sets bonding specific sysconfig options in given device map
+    #
+    # If any bonding specific option is present already it gets overwritten
+    # by new ones in case of collision. If any BONDING_SLAVEx from devmap
+    # is not set, then its value is set to 'nil'
+    #
+    # @param devmap [map] hash of a device's sysconfig variables
+    # @param slaves [array] list of strings, each string is a bond slave name
+    #
+    # @return updated device map
+    def setup_bonding(devmap, slaves, options)
+      raise ArgumentError, "Device map has to be provided." if devmap.nil?
+
+      slaves ||= []
+
+      slave_opts = devmap.select { |k, _| k =~ /BONDING_SLAVE/ }.keys
+      slave_opts.each { |s| devmap[s] = nil }
+      slaves.each_with_index { |s, i| devmap["BONDING_SLAVE#{i}"] = s }
+
+      devmap["BONDING_MODULE_OPTS"] = options || ""
+      devmap["BONDING_MASTER"] = "yes"
+
+      deep_copy(devmap)
```

Comment:
If you will return a copy, then it should be at the beginning instead of at the final, to not modify the devmap. 

So you can decide if return a modified copy of the given object or modify the object in place renaming the method to `setup_bonding!`